src/Features/CSharp/Portable/SignatureHelp/InvocationExpressionSignatureHelpProvider_MethodGroup.cs

```diff
@@ -58,17 +59,18 @@ internal partial class InvocationExpressionSignatureHelpProvider
                 methodGroup = methodGroup.Where(m => m.IsStatic);
             }
 
-            var accessibleMethods = methodGroup.Where(m => m.IsAccessibleWithin(within, throughTypeOpt: throughType)).ToList();
-            if (accessibleMethods.Count == 0)
+            var accessibleMethods = methodGroup.Where(m => m.IsAccessibleWithin(within, throughTypeOpt: throughType)).ToImmutableArrayOrEmpty();
+            if (accessibleMethods.Length == 0)
             {
-                return null;
+                return default;
             }
 
             var methodSet = accessibleMethods.ToSet();
-            accessibleMethods = accessibleMethods.Where(m => !IsHiddenByOtherMethod(m, methodSet)).ToList();
+            accessibleMethods = accessibleMethods.Where(m => !IsHiddenByOtherMethod(m, methodSet)).ToImmutableArrayOrEmpty();
 
-            return accessibleMethods.Select(m =>
-                ConvertMethodGroupMethod(m, invocationExpression, semanticModel, symbolDisplayService, anonymousTypeDisplayService, documentationCommentFormattingService, cancellationToken)).ToList();
+            return (accessibleMethods.Select(m =>
+                ConvertMethodGroupMethod(m, invocationExpression, semanticModel, symbolDisplayService, anonymousTypeDisplayService, documentationCommentFormattingService, cancellationToken)).ToList(),
+                TryGetSelectedIndex(accessibleMethods, currentSymbol));
```

Comment:
>TryGetSelectedIndex(accessibleMethods, currentSymbol) [](start = 16, length = 53)

:memo: if I understood the change correctly, this is the crux of the fix. We're calling `TryGetSelectedIndex` with `accessibleMethods` rather than `methodGroup`, so that the returned index is in bound of the relevant range (ie. that of `accessibleMethods`).

Reply:
Yup.